_posts/2021-05-26-registry-security.md

```diff
@@ -0,0 +1,64 @@
+* **Creator-only authorization option** - option to prevent changes to artifacts unless the authenticated user is the user who originally created the artifact
+
+
+Keycloak Configuration
+===
+
+### Authentication based on Keycloak
+In order to secure Apicurio Registry you will need a Keycloak server up and running along with some specific configuration.
+Essentially, you will need to create a Realm, two clients, and some roles (if you want to use roles).  By default Apicurio Registry expects the name of the Realm to be `registry`.
+When security is enabled, Apicurio Registry will expect two Keycloak clients.  By default, these clients should be named `registry-api` and `apicurio-registry` (but don't worry, you will be able to customize everything by using environment variables!)
+Below you have the required configuration for the clients.
+
+![Configure Apicurio Registry API_Client](/images/guides/registry-auth-client-api-config.png)
+
+![Configure Apicurio Registry_UI Client](/images/guides/registry-auth-client-ui-config.png)
+
+You will need to pay attention to the configuration - especially to the access types and the configuration URLs. 
+For the `registry-api` client, make sure to configure it as bearer-only. For the `apicurio-registry` client, you will need to configure it as public, to allow users to connect to your UI.
+You will also need to configure Web Origins, you can use `+` to allow Web Origins coming from the redirect URL.    
+
+Now, you will need to tweak the Apicurio Registry configuration to get this to work.
+
+![Configure Apicurio Registry Auth_Properties](/images/guides/registry-auth-properties-config.png)
+
+The first property that you will need to use is `AUTH_ENABLED`.  You will need to set it to true to enable the integration.  The second property you need to set is the Keycloak server url `KEYCLOAK_URL`.
+We provide some sensible defaults for most of the configuration, but please, note that you can customize most of them by using environment variables.
+
+### Role based authorization
+
+To enable roles, you need to set the `ROLES_ENABLED` propert to `true`. 
+
+![Configure Apicurio Registry Auth_Properties](/images/guides/registry-auth-http-paths-config.png)
+
+Of course, if you enable roles in Apicurio Registry, you will also need to create them in Keycloak. 
+You need to create them as Realm roles. The default roles expected by Apicurio Registry are `sr-admin`, `sr-developer`, and `sr-readonly`.
+When you enable roles, this will result in some HTTP paths and operations available just for certain roles. 
+You have the full configuration in the image above but, essentially, there will be some operations like those around rules, logs levels etc that are restricted to the admin role. 
+The developer role will be able to perform most of the operations and, obviously, the read-only role will be able just to read artifacts. 
```

Comment:
Suggest "The developer role can perform most of the operations and the read-only role can read or search artifacts."